pennylane/_grad.py

```diff
@@ -93,16 +94,25 @@ def _get_grad_fn(self, args):
         if len(argnum) == 1:
             argnum = argnum[0]
 
-        return self._grad_with_forward(
-            self._fun,
-            argnum=argnum,
-        )
+        return self._grad_with_forward(self._fun, argnum=argnum), argnum
 
     def __call__(self, *args, **kwargs):
         """Evaluates the gradient function, and saves the function value
         calculated during the forward pass in :attr:`.forward`."""
-        grad_value, ans = self._get_grad_fn(args)(*args, **kwargs)
+        grad_fn, argnum = self._get_grad_fn(args)
+
+        if not isinstance(argnum, int) and not argnum:
+            warnings.warn(
+                "Attempted to differentiate a function with no trainable parameters. "
+                "If this is unintended, please add trainable parameters via the "
+                "`requires_grad` attribute or `argnum` keyword."
```

Comment:
```suggestion
                "requires_grad attribute or argnum keyword."
```
since the warning is not rendered :)

Reply:
Ah it thought it would be useful still to delineate the names from the warning text. Maybe regular single quotes then?

Reply:
sure! my default is always to: check what NumPy does and follow the same style :)

Reply:
That is a good guide 😅 I checked other PennyLane messages and single quotes are used quite a bit.